--- tools/install.py
+++ tools/install.py
@@ -140,10 +140,10 @@ def subdir_files(path, dest, action):
   for subdir, files_in_path in ret.items():
     action(files_in_path, subdir + '/')

-def files(action):
+def files(action,build_type):
   is_windows = sys.platform == 'win32'
   output_file = 'node'
-  output_prefix = 'out/Release/'
+  output_prefix = 'out/{0}/'.format(build_type)

   if is_windows:
     output_file += '.exe'
@@ -321,10 +321,11 @@ def run(args):
   # argv[3] is a custom install prefix (think PREFIX)
   # Difference is that dst_dir won't be included in shebang lines etc.
   dst_dir = args[2] if len(args) > 2 else ''
-
+  build_type = "Release"
   if len(args) > 3:
     node_prefix = args[3]
-
+  if len(args) > 4:
+    build_type = args[4]
   # install_path thus becomes the base target directory.
   install_path = dst_dir + node_prefix + '/'

@@ -339,10 +340,10 @@ def run(args):
       return
   else:
     if cmd == 'install':
-      files(install)
+      files(install, build_type)
       return
     if cmd == 'uninstall':
-      files(uninstall)
+      files(uninstall, build_type)
       return

   raise RuntimeError('Bad command: %s\n' % cmd)
